synapse/rest/client/v1/room.py

```diff
@@ -171,6 +172,77 @@ def on_PUT(self, request, room_id, event_type, state_key, txn_id=None):
         if state_key is not None:
             event_dict["state_key"] = state_key
 
+        # If the event is a retention policy definition for the room, check that it
+        # respects the boundaries imposed by the server's administrator.
+        if event_type == EventTypes.Retention:
+            min_lifetime = content.get("min_lifetime")
```

Comment:
could you factor all of this (within the `if` block) out to a separate function?

Also: I think it should happen within EventValidator.validate_new, which is where similar checks happen on other event types.